sdk/client-core/core/src/main/java/io/clientcore/core/util/ServerSentEventUtils.java

```diff
@@ -20,114 +22,121 @@
-        StringBuilder collectedData = new StringBuilder();
         ServerSentEvent event = null;
+
         try {
+            StringBuilder collectedData = new StringBuilder();
             String line;
+
             while ((line = reader.readLine()) != null) {
                 collectedData.append(line).append("\n");
+
                 if (isEndOfBlock(collectedData)) {
                     event = processLines(collectedData.toString().split("\n"));
+
                     if (!Objects.equals(event.getEvent(), DEFAULT_EVENT) || event.getData() != null) {
                         listener.onEvent(event);
                     }
+
                     collectedData = new StringBuilder(); // clear the collected data
                 }
             }
+
             listener.onClose();
         } catch (IOException e) {
-            return new RetrySSEResult(e,
-                    event != null ? event.getId() : -1,
-                    event != null ? ServerSentEventHelper.getRetryAfter(event) : null);
+            return new RetrySSEResult(e, event != null ? event.getId() : -1,
+                event != null ? ServerSentEventHelper.getRetryAfter(event) : null);
         }
+
         return null;
     }
 
     /**
-     * Retries the request if the listener allows it
+     * Retries the {@link HttpRequest} if the listener allows it.
      *
-     * @param retrySSEResult  the result of the retry
-     * @param listener The listener object attached with the httpRequest
-     * @param httpRequest the HTTP Request being sent
+     * @param retrySSEResult The {@link RetrySSEResult result} of the retry.
+     * @param listener The listener object attached with the {@link HttpRequest}.
+     * @param httpRequest The {@link HttpRequest} to send.
      */
-    private static boolean retryExceptionForSSE(RetrySSEResult retrySSEResult, ServerSentEventListener listener, HttpRequest httpRequest) {
-        if (Thread.currentThread().isInterrupted()
-                || !listener.shouldRetry(retrySSEResult.getException(),
-                retrySSEResult.getRetryAfter(),
-                retrySSEResult.getLastEventId())) {
+    private static boolean retryExceptionForSSE(RetrySSEResult retrySSEResult, ServerSentEventListener listener,
+                                                HttpRequest httpRequest) {
+        if (Thread.currentThread().isInterrupted() || !listener.shouldRetry(retrySSEResult.getException(),
+            retrySSEResult.getRetryAfter(), retrySSEResult.getLastEventId())) {
+
             listener.onError(retrySSEResult.getException());
+
             return true;
         }
 
         if (retrySSEResult.getLastEventId() != -1) {
             httpRequest.getHeaders()
-                    .add(HttpHeaderName.fromString(LAST_EVENT_ID), String.valueOf(retrySSEResult.getLastEventId()));
+                .add(HttpHeaderName.fromString(LAST_EVENT_ID), String.valueOf(retrySSEResult.getLastEventId()));
```

Comment:
Really should make that `HttpHeaderName` a constant instead of needing to look it up each time